mayday/journal.go

```diff
@@ -0,0 +1,79 @@
+package mayday
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/coreos/go-systemd/dbus"
+	"log"
+	"os/exec"
+	"regexp"
+	"time"
+)
+
+type Journal struct {
+	Name    string
+	Content *bytes.Buffer // the contents of the log, populated by Get()
+}
+
+func GetJournals() ([]Journal, error) {
+	var svcs []Journal
+
+	c, err := dbus.New()
+	if err != nil {
+		return svcs, err
+	}
+
+	defer c.Close()
+
+	units, err := c.ListUnits()
+	if err != nil {
+		return svcs, err
+	}
+
+	pathre := regexp.MustCompile(`/usr/lib(32|64)?/systemd/system/.*\.service`)
```

Comment:
Everything in `journal.go` was just factored out of [the original `/mayday.go`](https://github.com/coreos/mayday/blob/ad98a064325429099a5dfd763c6e035a094dfd18/mayday.go).